Document for the C/C++ samples (#1442)
microsoft/onnxruntime: samples/c_cxx/imagenet/README.md

If you're going to train the model by yourself, then you need the full dataset, which is about 500GB. Otherwise, you only need the 
validation data set, which is just about 3GB.

For how to get the data, see [ImageNet Download faq](http://image-net.org/download-faq). Once you get an account, visit http://www.image-net.org/download-images. You will find "Download links to ILSVRC2012 image data" on that page

And also, please download the "[imagenet_lsvrc_2015_synsets.txt](https://raw.githubusercontent.com/tensorflow/models/master/research/slim/datasets/imagenet_lsvrc_2015_synsets.txt)" and "[imagenet_2012_validation_synset_labels.txt](https://raw.githubusercontent.com/tensorflow/models/master/research/slim/datasets/imagenet_2012_validation_synset_labels.txt)" from tensorflow models repo.

# Get the model
Please check [https://github.com/tensorflow/models/tree/master/research/slim/](https://github.com/tensorflow/models/tree/master/research/slim/).
You may either train the model by yourself, or just download a pretrained model provided by Google. 

> If you don't know which one to download and try, we suggest you may choose the [Inception V4](http://download.tensorflow.org/models/inception_v4_2016_09_09.tar.gz) model as a start point.

Reviewer: ```suggestion
If you don't know which one to download and try, we suggest you choose the [Inception V4](http://download.tensorflow.org/models/inception_v4_2016_09_09.tar.gz) model as a starting point.
```

After downloading, please uncompress it.
```
tar -zxvf inception_v4_2016_09_09.tar.gz
```

The [Inception V4] zip file only contains a single checkpoint file: inception_v4.ckpt. It can't be directly used for inferencing. 

> You need the network definition and combine it with the checkpoint. Please follow the steps below:

Reviewer: ```suggestion
You need to combine the network definition and the checkpoint. Please follow the steps below:
```

1. Export the graph
Create an new folder. At there, execute
```
git clone https://github.com/tensorflow/models .
python research\slim\export_inference_graph.py --model_name=inception_v4 --output_file=grpah.pb
```

2. Freeze the graph
Run
```
freeze_graph.exe --input_graph=graph.pb --input_checkpoint=inception_v4.ckpt --output_graph=inception_v4.pb --output_node_names=InceptionV4/Logits/Predictions --input_binary=true
```

# Convert the model to ONNX

```
pip install --upgrade tf2onnx 
python -m tf2onnx.convert --input inception_v4.pb --inputs input:0 --outputs InceptionV4/Logits/Predictions:0 --opset 10 --output inception_v4.onnx
```

# Run the inferencing 

> In your build dir of onnxruntime_samples, search "image_classifier.exe" and run 

Reviewer: ```suggestion
In your build dir of onnxruntime_samples, search for "image_classifier.exe" and run 
```

```
image_classifier.exe C:\tools\imagnet_validation_data inception_v4.onnx imagenet_lsvrc_2015_synsets.txt imagenet_2012_validation_synset_labels.txt 32
```

> Please replace the file names with corresponding file paths.

Reviewer: ```suggestion
Please replace the file names with the corresponding file paths.
```
